diff --git a/dist/index.d.mts b/dist/index.d.mts
new file mode 100644
index 0000000000000000000000000000000000000000..f0c7efeb567872dc142b5a283b60ebb72927793f
--- /dev/null
+++ b/dist/index.d.mts
@@ -0,0 +1,42 @@
+import { Plugin } from 'vite';
+
+/**
+ * Plugin options.
+ */
+interface Options {
+    /**
+     * Relative path to the directory to search for page components.
+     * @default 'src/layouts'
+     */
+    layoutsDirs: string | string[];
+    /**
+     * Valid file extensions for page components.
+     * @default ['vue']
+     */
+    extensions: string[];
+    /**
+     * List of path globs to exclude when resolving pages.
+     */
+    exclude: string[];
+    /**
+     * Filename of default layout (".vue" is not needed)
+     * @default 'default'
+     */
+    defaultLayout: string;
+    /**
+     * Mode for importing layouts
+     */
+    importMode: (name: string) => 'sync' | 'async';
+}
+type FileContainer = {
+    path: string;
+    files: string[];
+};
+type UserOptions = Partial<Options>;
+interface ResolvedOptions extends Options {
+}
+
+declare function defaultImportMode(name: string): "sync" | "async";
+declare function layoutPlugin(userOptions?: UserOptions): Plugin;
+
+export { FileContainer, ResolvedOptions, UserOptions, layoutPlugin as default, defaultImportMode };
diff --git a/package.json b/package.json
index de999023f87c55dee2e57e2583e1e96b6f95095f..c6cb7906d7dd91c7ac9b7f98b6375b607708ce22 100644
--- a/package.json
+++ b/package.json
@@ -19,8 +19,14 @@
   ],
   "exports": {
     ".": {
-      "require": "./dist/index.js",
-      "import": "./dist/index.mjs"
+      "import": {
+        "types": "./dist/index.d.mts",
+        "default": "./dist/index.mjs"
+      },
+      "require": {
+        "types": "./dist/index.d.ts",
+        "default": "./dist/index.js"
+      }
     },
     "./*": "./*"
   },